stem/model.py

```diff
@@ -1865,9 +1867,12 @@ def __finalise_json_output(self, input_folder: str):
                 new_json = {key: value for key, value in json_data_tmp.items() if "NODE" not in key}
 
                 # adjust the nodal outputs in the right order
-                for node_id in output_model_part.mesh.nodes.keys():
+                for node_id, node in output_model_part.mesh.nodes.items():
                     node_key = f"NODE_{node_id}"
-                    new_json[node_key] = json_data_tmp[node_key]
+                    data_node = json_data_tmp[node_key]
+                    # reassign the corresponding nodal outputs including the nodal coordinates at the top
+                    new_json[node_key] = {'COORDINATES': node.coordinates}
+                    new_json[node_key].update(data_node)
```

Comment:
i think you can do somehting like:

`new_json[node_key]= {"COORDINATES": node.coordinates | json_data_tmp[node_key]`

Reply:
but keep the comment, such that it is clear that this is done to have the coordinates at the top

Reply:
this is valid, from 3.9. Since we dropped 3.8 we can use this